internal/testutils/test_utils.go

```diff
@@ -0,0 +1,42 @@
+// Package testutils provides some utilities that may be useful for testing
+package testutils
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"github.com/stretchr/testify/mock"
+
+	"go.infratographer.com/permissions-api/pkg/permissions/mockpermissions"
+)
+
+// MockPermissions creates a context from the given context with mocks for permission-api methods
+func MockPermissions(ctx context.Context) context.Context {
+	// mock permissions
+	perms := new(mockpermissions.MockPermissions)
+	perms.On("CreateAuthRelationships", mock.Anything, mock.Anything, mock.Anything).Return(nil)
+	perms.On("DeleteAuthRelationships", mock.Anything, mock.Anything, mock.Anything).Return(nil)
+
+	ctx = perms.ContextWithHandler(ctx)
+
+	return ctx
+}
+
+// IfErrPanic conditionally panics on err with msg
+func IfErrPanic(msg string, err error) {
+	if err != nil {
+		log.Panicf("%s err: %s", msg, err.Error())
+	}
+}
```

Comment:
for all these error checks, you can use `require.NoError(t, err, msg)`

Reply:
I think it definitely makes sense for the stuff in the tests, but the teardown/setup exist outside the tests so can't replace those ones.